### DISCUSSION TITLE
Help with Audio Effects - Storm Trooper Voice Changer

### DISCUSSION THREAD

**Discussion by cmcooper1971** [pschatzmann/arduino-audio-tools#1229] [Q&A] [answered]
Hi All,

I have successfully built a I2S Digital Amplifier, which consists of an SD card reader, an I2S Microphone (INMP441) linked to an I2S Texas Instruments Class D Amplifier (TAS5760L), all of which is controlled using Phils, Arduino Audio Tools library.

This is so a friend can either select a variety of tracks from the Star Wars films, or speak as he walks around his Comic Cons, please see:

https://github.com/pschatzmann/arduino-audio-tools/assets/63595661/17796bc1-e089-468b-8be3-2b95cbb418ec

I wish to improve the voice amplifier and add Audio Effects so he sounds like a Storm Trooper when speaking. I have been trying to find out what Audio Effects I would need to apply to the input microphone (or output)

Does anyone have any ideas which effects to apply?

Anyone who is interested, this is my current build which all works well:

`AudioInfo info(44100, 2, 16);`

`I2SStream in;`
`VolumeStream volume(in);`
`I2SStream out;`
`EncodedAudioStream decoder(&out, new WAVDecoder()); // Decoding stream`
`EncodedAudioStream decoder(&out, new MP3DecoderHelix()); // Decoding stream`
`StreamCopy mp3;`
`StreamCopy mic(out, in);`

Setup config looks like this:

`// Setup I2S in`

`AudioLogger::instance().begin(Serial, AudioLogger::Warning);`

`debugln("Starting I2S...");`

`auto config_in = in.defaultConfig(RX_MODE);`
`config_in.copyFrom(info);`
`config_in.i2s_format = I2S_STD_FORMAT;`
`config_in.is_master = true;`
`config_in.port_no = 0;`
`config_in.pin_bck = I2S_BCLK_0;		// SCK`
`config_in.pin_ws = I2S_LRC_0;			// WS or LRCLK or LRCK or FS`
`config_in.pin_data = I2S_DIN_0;		// SD or SDIN or SDOUT`
`//config_in.pin_mck = I2S_MCLK_1;`

`in.begin(config_in);`

`// Setup I2S out`

`// effects.addEffect(pitchEffect);`

`auto config_out = out.defaultConfig(TX_MODE);`
`config_out.copyFrom(info);`
`config_out.i2s_format = I2S_STD_FORMAT;`
`config_out.is_master = true;`
`config_out.port_no = 1;`
`config_out.pin_bck = I2S_BCLK_1;		// SCK`
`config_out.pin_ws = I2S_LRC_1;		// WS or LRCLK or LRCK or FS`
`config_out.pin_data = I2S_DOUT_1;	// SD or SDIN or SDOUT`
`config_out.pin_mck = I2S_MCLK_1;`

`out.begin(config_out);`
`//effects.begin(config_out);`

`// Setup I2S based on sampling rate provided by decoder`

`decoder.setNotifyAudioChange(out);`
`decoder.begin();`

`// Begin mp3`

`mp3.begin(decoder, audioFile);`

`// Load initial audio file into I2S`

`audioFile = SD.open("/0001.mp3");`

I then enable and disable using a variety of controls, such as Single Tap of the unit to switch between MP3 and Microphone inputs.

Regards.

Christopher

**Comment by pschatzmann** [comment, marked as answer]
Not sure what effect you are looking for, but I suggest that you look into [Pitch Shifting](https://github.com/pschatzmann/arduino-audio-tools/wiki/Pitch-Shifting).

You could also try to apply a low pass filter to get a robotic effect...

**Reply by cmcooper1971** [reply]
Thanks Phil - I will give it a try.

**Reply by weow147** [reply]
Did this work for you Sir?  I'm interested in implement a 'robot' effect, but storm trooper would be pretty cool as well!
